Adds metadata components

cartogram opened the pull request:
This PR adds a few new components for rendering metadata to the `react-html` package.

The new components are:
- `<AppleHomeScreen />`
- `<Responsive />`

cartogram commented on README.md:
```diff
@@ -34,6 +34,7 @@ Each package has its own `README` and documentation describing usage.
 | koa-metrics | [directory](packages/koa-metrics) | [![npm version](https://badge.fury.io/js/%40shopify%2Fkoa-metrics.svg)](https://badge.fury.io/js/%40shopify%2Fkoa-metrics) |
 | koa-shopify-auth | [directory](packages/koa-shopify-auth) | [![npm version](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-auth.svg)](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-auth) |
 | koa-shopify-graphql-proxy | [directory](packages/koa-shopify-graphql-proxy) | [![npm version](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-graphql-proxy.svg)](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-graphql-proxy) |
+| koa-shopify-webhooks | [directory](packages/koa-shopify-webhooks) | [![npm version](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-webhooks.svg)](https://badge.fury.io/js/%40shopify%2Fkoa-shopify-webhooks) |
```
Unrelated change.

TheMallen commented on packages/react-html/README.md:
```diff
@@ -230,6 +230,18 @@ Renders a `<link />` tag with the necessary props to specify a favicon. Accepts
 
 Renders a `<link />` tag that preconnects the browser to the host specified by the `source` prop. You can read more about preconnecting on [Google’s guide to resource prioritization](https://developers.google.com/web/fundamentals/performance/resource-prioritization#preconnect).
 
+### `<Baseline />`
```
I don't really like this name, especially since we have the concept of `baseline` bundles in sewing-kit. Actually I'm not sure we want this to be a component. I sort of feel like the basic meta tags should just always be rendered by `HTML` and then `title` and `favicon` can either be used directly or passed as props to `HTML` itself.

lemonmade replied:
I like Mal's idea here 👍

TheMallen commented on packages/react-html/README.md:
```diff
@@ -230,6 +230,18 @@ Renders a `<link />` tag with the necessary props to specify a favicon. Accepts
+
+Renders a minimal set of `<Meta />` tags to specify basic app information. Accepts a `title` property that is passed as the children of a `<Title />` component, and a `favicon` source that is passed to a `<Favicon />` component.
+
+### `<Responsive />`
```
I'm also not sure we ever don't want these

lemonmade replied:
This one I think is good, the incantation to get the right meta tags is always a pain in the but IMO

TheMallen replied:
Yeah, I just mean do we ever want to not have this? Why not just have the component always include them

lemonmade commented on packages/react-html/src/components/tests/AppleHomeScreen.test.tsx:
```diff
@@ -0,0 +1,73 @@
+    const appleHomeScreen = mount(<AppleHomeScreen />);
+
+    expect(
+      findMeta(appleHomeScreen.find(Meta), {
```
Feels a bit awkward to have `findMeta(thing.find(Meta)`, maybe just `findMeta(thing, `?

cartogram commented:
@TheMallen Updated:
1. removed `<Baseline />` altogether and merged with `<HTML />` using the title prop as a fallback if one was not extracted from the page.
2. fixed up the `findMeta` utility and moved it into the `AppleHomeScreen` component file since that is the only place it is used (since I remove `<Baseline />`)
